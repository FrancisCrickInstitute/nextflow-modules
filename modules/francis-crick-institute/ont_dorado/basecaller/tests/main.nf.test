nextflow_process {

    name "Test Module ONT_DORADO_BASECALLER"
    script "../main.nf"
    process "ONT_DORADO_BASECALLER"

    tag "modules"
    tag "ont"
    tag "ont_dorado"
    tag "ont_dorado/basecaller"

    test("pod5") {

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id:'test' ],
                    file('$baseDir/' + params.test_data_dir + '/ont/seq_runs/pod5_100/pod5/PAW36768_7b0e5258_08c7ccb8_0_sub100.pod5', checkIfExists: true)
                ])
                input[1] = []
                input[2] = 'hac'
                input[3] = []
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out.versions).match()}
            )
        }
    }
}
