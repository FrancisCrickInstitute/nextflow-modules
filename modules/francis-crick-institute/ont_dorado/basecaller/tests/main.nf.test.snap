{
    "pod5": {
        "content": [
            [
                "versions.yml:md5,513856826291b2061d33db76a43805aa"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-29T16:11:51.011485"
    }
}